### DISCUSSION TITLE
Data Model Templates

### DISCUSSION THREAD

**Discussion by ernst86913** [directus/directus#11535] [Feature Requests]
@benhaynes hi Ben - are there any plans in the future to have data model templates as extensions?

example: many people will need a „contacts“ data model for customers, partners,…. Or…. Many will need invoice data model… or products data model,…

currently building this is left to each and every extension module. This will result in a huge amount of different models - and will prevent extension modes working with each other.

better idea would be to have standard data models as extensions which can be used - and then provide the possibility to extend those on an individual basis.

I am not suggesting that this should be part of the core product… but - it finally would result in a list of data model extensions which can be improved in the community in a standardized way…

Hope that makes sense… what do you think strategically about it?

KR
Roman

**Comment by benhaynes** [comment]
Definitely. Still thinking through the name of this extension type (schema, template, project template, boilerplate, etc). We'd likely keep this as _just_ the data model, and then you can use the "Bundle" extension type to group one of these "Project Templates" with the interfaces, layouts, and other extensions that it uses/references. In that way, you could have something like a "Task Management" bundle that has a `tasks` table with appropriate fields configured to use custom interfaces/displays/layouts... eg a Kanban Layout.

The thinking is that these would be incrementally adoptable, so you could pick and choose "Task Management" + "Data Analytics" + "Application Performance Management" and build out a full project based on existing pieces.

**Reply by ernst86913** [reply]
@benhaynes  Great. Exactly my thinking. Thanks for quick response

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
